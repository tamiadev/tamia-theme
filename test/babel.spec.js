const babel = require('babel-core');
const plugin = require('../babel');

const compile = source => babel.transform(source, { plugins: [plugin] }).code;

test('do not touch unrelated code', () => {
	const result = compile(`
import styled from "react-emotion";
import theme from "./test/fixtures/theme";
const a = theme.space.xl;
const Container = styled("div")\`
  color: deepskyblue;
  margin-bottom: \${a};
\`;
	`);
	expect(result).toMatchSnapshot();
});

test('inline theme variables', () => {
	const result = compile(`
import styled from "react-emotion";
import theme from "./test/fixtures/theme";
const Container = styled("div")\`
  color: deepskyblue;
  margin-bottom: \${theme.space.xl};
\`;
	`);
	expect(result).toMatchSnapshot();
});

test('inline multiple theme variables', () => {
	const result = compile(`
import styled from "react-emotion";
import theme from "./test/fixtures/theme";
const Container = styled("div")\`
  color: deepskyblue;
  margin-top: \${theme.space.s};
  margin-bottom: \${theme.space.m};
  margin-left: \${theme.space.l};
  margin-right: \${theme.space.xl};
\`;
	`);
	expect(result).toMatchSnapshot();
});

test('works with any level of nesting', () => {
	const result = compile(`
import styled from "react-emotion";
import theme from "./test/fixtures/theme";
const Container = styled("div")\`
  color: deepskyblue;
  font-family: \${theme.font};
  margin: \${theme.space.l};
  padding: \${theme.foo.bar.baz};
\`;
	`);
	expect(result).toMatchSnapshot();
});

test('throw when theme file not found', () => {
	const fn = () =>
		compile(`
import styled from "react-emotion";
import theme from "./pizza";
const Container = styled("div")\`
  margin: \${theme.pizza.salami};
\`;
	`);
	expect(fn).toThrowError('Cannot find module');
});

test('throw when value not found', () => {
	const fn = () =>
		compile(`
import styled from "react-emotion";
import theme from "./test/fixtures/theme";
const Container = styled("div")\`
  margin: \${theme.pizza.salami};
\`;
	`);
	expect(fn).toThrowError('Value of theme.pizza.salami is undefined');
});
