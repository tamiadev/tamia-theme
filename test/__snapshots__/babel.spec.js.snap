// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`do not touch unrelated code 1`] = `
"
import styled from \\"react-emotion\\";
import theme from \\"./test/fixtures/theme\\";
const a = theme.space.xl;
const Container = styled(\\"div\\")\`
  color: deepskyblue;
  margin-bottom: \${a};
\`;"
`;

exports[`inline multiple theme variables 1`] = `
"
import styled from \\"react-emotion\\";
import theme from \\"./test/fixtures/theme\\";
const Container = styled(\\"div\\")\`
  color: deepskyblue;
  margin-top: 8px;
  margin-bottom: 16px;
  margin-left: 32px;
  margin-right: 64px;
\`;"
`;

exports[`inline theme variables 1`] = `
"
import styled from \\"react-emotion\\";
import theme from \\"./test/fixtures/theme\\";
const Container = styled(\\"div\\")\`
  color: deepskyblue;
  margin-bottom: 64px;
\`;"
`;

exports[`works with any level of nesting 1`] = `
"
import styled from \\"react-emotion\\";
import theme from \\"./test/fixtures/theme\\";
const Container = styled(\\"div\\")\`
  color: deepskyblue;
  font-family: Comic Sans;
  margin: 32px;
  padding: 42;
\`;"
`;
